plugins/modules/s3_object.py

```diff
@@ -494,12 +495,14 @@ def bucket_check(module, s3, bucket, validate=True):
     except is_boto3_error_code('404'):
         return False
     except is_boto3_error_code('403') as e:  # pylint: disable=duplicate-except
-        if validate is True:
-            module.fail_json_aws(e, msg="Failed while looking up bucket (during bucket_check) %s." % bucket)
+        if validate:
+            module.fail_json_aws(e,
+                msg="Failed while looking up bucket (during bucket_check) %s." % bucket)
     except botocore.exceptions.EndpointConnectionError as e:  # pylint: disable=duplicate-except
         module.fail_json_aws(e, msg="Invalid endpoint provided")
     except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:  # pylint: disable=duplicate-except
-        module.fail_json_aws(e, msg="Failed while looking up bucket (during bucket_check) %s." % bucket)
+        module.fail_json_aws(e,
+            msg="Failed while looking up bucket (during bucket_check) %s." % bucket)
```

Comment:
`exists` is never changed. You can just drop the variable and end with `return True`.